server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -2369,7 +2377,7 @@ public void taskSucceeded(ClusterStateUpdateTask clusterStateTaskListener, Void
                     public ClusterState execute(ClusterState currentState) {
                         assertEquals(priority, prioritiesQueue.poll());
                         assertEquals(priority, priority());
-                        return currentState;
+                        return randomBoolean() ? currentState : ClusterState.builder(currentState).build();
```

Comment:
We're using a single-threaded executor here, so by sometimes triggering a publication and sometimes forking the publication handling onto a separate thread we demonstrate that there is no longer any blocking involved in the publication process.